modules/core/include/opencv2/core/mat.hpp

```diff
@@ -61,8 +61,9 @@ namespace cv
 //! @addtogroup core_basic
 //! @{
 
-enum { ACCESS_READ=1<<24, ACCESS_WRITE=1<<25,
-    ACCESS_RW=3<<24, ACCESS_MASK=ACCESS_RW, ACCESS_FAST=1<<26 };
+enum AccessFlag { ACCESS_READ=1<<24, ACCESS_WRITE=1<<25,
+    ACCESS_RW=3<<24, ACCESS_MASK=ACCESS_RW, ACCESS_FAST=1<<26,
+    CV_FLAGS_ENUM = ACCESS_READ };
 
```

Comment:
As you can see, `CV_FLAGS_ENUM` is introduced to enable bitwise operators, and it can take any value, and in this case, it is assigned `ACCESS_READ` for safety against ignorant users.

Reply:
Cool. I think that would work.

The obvious disadvantage is that we have extra `CV_FLAGS_ENUM` in every flags enum and in documentation etc. Also in my experience widely used SFINAE can affect compilation times pretty badly.

Therefore, I would prefer the solution above, but this is certainly workable too. If you want to go with this, I suggest to add constexpr to allow using it in constant expressions.

Reply:
`constexpr` is available from Visual Studio 2015.
Do we really want to drop support of Visual Studio 2013? It is still widely used...

Reply:
I think OpenCV 4 will be C++11 as per [OE-4 ](https://github.com/opencv/opencv/wiki/OE-4.-OpenCV-4)